docs/react.md

```diff
@@ -0,0 +1,214 @@
+});
+
+// Subscribe to member enter events only
+useMembers('enter', (memberJoined) => {
+  console.log(memberJoined);
+});
+
+// Subscribe to member leave events only
+useMembers('leave', (memberLeft) => {
+  console.log(memberLeft);
+});
+
+// Subscribe to member remove events only
+useMembers('remove', (memberRemoved) => {
+  console.log(memberRemoved);
+});
+
+// Subscribe to profile updates on members only
+useMembers('updateProfile', (memberProfileUpdated) => {
+  console.log(memberProfileUpdated);
+});
+
+// Subscribe to all updates to members
+useMembers('update', (memberUpdate) => {
+  console.log(memberUpdate);
+});
+```
+
+### useLocation
+
+The `useLocation` hook lets you subscribe to location events.
+Location events are emitted whenever a member changes location.
+
+```javascript
+useLocation((locationUpdate) => {
+  console.log(locationUpdate);
+});
+```
+
+it also lets you update current member location by using `update` method provided by hook:
+
+```javascript
+const { update } = useLocation((locationUpdate) => {
+  console.log(locationUpdate);
+});
+```
+
+### useLocks
+
+The `useLocks` hook lets you subscribe to lock events by registering a listener.
+Lock events are emitted whenever the lock state transitions into `locked` or `unlocked`.
+
+```javascript
+useLocks((lockUpdate) => {
+  console.log(lockUpdate);
+});
+```
+
+### useLock
+
+The `useLock` returns the status of a lock and, if it has been acquired, the member holding the lock
+
+```javascript
+const { status, member } = useLock('my-lock');
+```
+
+### useCursors
+
+The `useCursors` hook allows you to track a member's pointer position and provide an updated view for all members across an application:
+
+```javascript
+// Subscribe to events published on "mousemove" by all members
+const { set } =  useCursors((cursorUpdate) => {
+  console.log(cursorUpdate);
+});
+
+useEffect(() => {
+  // Publish a your cursor position on "mousemove" including optional data
+  const eventListener = ({ clientX, clientY }) => {
+    set({ position: { x: clientX, y: clientY }, data: { color: 'red' } });
+  }
+
+  window.addEventListener('mousemove', eventListener);
+
+  return () => {
+    window.removeEventListener('mousemove', eventListener);
+  };
+});
+```
+
+If you provide `{ returnCursors: true }` as an option you can get active members cursors:
+
+```javascript
+const { cursors } =  useCursors((cursorUpdate) => {
+  console.log(cursorUpdate);
+}, { returnCursors: true });
+```
+
+---
+
+### Error Handling
+
+When using the Spaces react hooks, your Ably client may encounter a variety of errors, for example if it doesn't have permissions to attach to a channel it may encounter a channel error, or if it loses connection from the Ably network it may encounter a connection error.
+
+To allow you to handle these errors, the `useSpace`, `useMembers`, `useLocks`, `useCursors` hooks return connection and channel errors so that you can react to them in your components:
```

Comment:
```suggestion
`useSpace`, `useMembers`, `useLocks` and `useCursors` return connection and channel errors you may encounter, so that you can handle then within your components. This may include when a client doesn't have permission to attach to a channel, or if it loses its connection to Ably.
```